checker/raw_result.go

```diff
@@ -283,24 +283,27 @@ type CIIBestPracticesData struct {
 	Badge CIIBadge
 }
 
+// DangerousWorkflowType represents a type of dangerous workflow.
+type DangerousWorkflowType int
+
+const (
+	// DangerousWorkflowScriptInjection represents a script injection.
+	DangerousWorkflowScriptInjection DangerousWorkflowType = iota
+	// DangerousWorkflowUntrustedCheckout represents an untrusted checkout.
+	DangerousWorkflowUntrustedCheckout
+)
+
 // DangerousWorkflowData contains raw results
 // for dangerous workflow check.
 type DangerousWorkflowData struct {
-	ScriptInjections   []ScriptInjection
-	UntrustedCheckouts []UntrustedCheckout
-	// TODO: other
-}
-
-// UntrustedCheckout represents an untrusted checkout.
-type UntrustedCheckout struct {
-	Job  *WorkflowJob
-	File File
+	Workflows []Workflow
 }
 
-// ScriptInjection represents a script injection.
-type ScriptInjection struct {
+// Workflow represents a result for a workflow.
+type Workflow struct {
 	Job  *WorkflowJob
 	File File
+	Type DangerousWorkflowType
```

Comment:
Maybe rename `Type` -> `DetectedDangerousPattern` (or something similar) to make it clear what this field represents?

Also, should this be part of `WorkflowJob` since each job can have it's own unique dangerous pattern?

Reply:
> Maybe rename `Type` -> `DetectedDangerousPattern` (or something similar) to make it clear what this field represents?

I'll update the structure from `Workflow` to `DangerousWorkflow` instead.

> Also, should this be part of `WorkflowJob` since each job can have it's own unique dangerous pattern?

`WorkflowJob` should be generic so that we can re-use it for Token-Permisison and Pinned-Dependencies check, o I think `Type` should not be part of it.

Reply:
How do we plan to surface information about multiple dangerous patterns in a workflow?

Reply:
using the Type:
```
type DangerousWorkflowData struct {
	Workflows []DangerousWorkflow
}

// DangerousWorkflow represents a dangerous workflow.
type DangerousWorkflow struct {
	Workflow Workflow
	Type     DangerousWorkflowType
}

// Workflow is generic and re-usable.
type Workflow struct {
	Job  *WorkflowJob
	File File
}

```

Wdut?

Reply:
Makes sense. LGTM.